drivers/driver_linux.go

```diff
@@ -165,8 +166,22 @@ func (c *defaultChecker) IsMounted(path string) bool {
 // Mounted checks if the given path is mounted as the fs type
 func Mounted(fsType FsMagic, mountPath string) (bool, error) {
 	var buf unix.Statfs_t
+	var bufParent unix.Statfs_t
+
 	if err := unix.Statfs(mountPath, &buf); err != nil {
 		return false, err
 	}
-	return FsMagic(buf.Type) == fsType, nil
+	if FsMagic(buf.Type) != fsType {
+		return false, nil
+	}
+	// it is already the root
+	if mountPath == "/" {
+		return true, nil
+	}
+	if err := unix.Statfs(filepath.Dir(mountPath), &bufParent); err != nil {
+		return true, err
+	}
+	// check that the parent directory is on a different mount
+	// than mountPath
+	return bufParent.Fsid != buf.Fsid, nil
```

Comment:
I like the idea of comparing information about the two directories, but the kernel currently always returns an fsid of 0 for fuse filesystems, so this comparison will always be `false`.  The `st_dev` value returned when we `syscall.Stat()` the directories, though, should be unique for different FUSE mounts.

Reply:
good idea, I've amended the patch